cmake/templates/HPXConfig.cmake.in

```diff
@@ -43,8 +41,15 @@ set(HPX_Fortran_COMPILER "@CMAKE_Fortran_COMPILER@" CACHE STRING "Fortran compil
 set(HPX_Fortran_COMPILER_ID "@CMAKE_Fortran_COMPILER_ID@" CACHE STRING "Fortran compiler id for HPX" FORCE)
 set(HPX_Fortran_COMPILER_VERSION "@CMAKE_Fortran_COMPILER_VERSION@" CACHE STRING "Fortran compiler version for HPX" FORCE)
 
+# Setup the imported libraries (publicly linked)
 set(HPX_BOOST_ROOT "@BOOST_ROOT@")
-set(HPX_BOOST_LIBRARIES "@Boost_LIBRARIES@")
+# Separate targets to be unarily linked to some modules
+include(HPX_SetupBoost)
+include(HPX_SetupBoostFilesystem)
+include(HPX_SetupBoostProgramOptions)
```

Comment:
Are the variables for enabling and disabling filesystem and program options set before these are included?